mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -150,6 +150,26 @@ sealed trait Vector extends Serializable {
       toDense
     }
   }
+
+  /**
+   * Find the index of a maximal element.  Returns the first maximal element in case of a tie.
+   * Returns -1 if vector has length 0.
+   */
+  private[spark] def findMax: Int = {
+    if (size == 0) {
+      0
```

Comment:
`-1`

We can also save `if .... else`:

~~~scala
var maxIdx = -1
var maxValue = Double.NegativeInfinity
foreachActive { (idx, value) =>
  ...
~~~